packages/hardhat-core/src/internal/hardhat-network/provider/vm/vm-adapter.ts

```diff
@@ -53,6 +53,7 @@ export interface VMAdapter {
 
   // setters
   putAccount(address: Address, account: Account): Promise<void>;
+  putBlock(block: Block): Promise<void>;
```

Comment:
If I'm understanding correctly, this method was added because the VM needs access to the last block hashes (for the `BLOCKHASH` opcode). But I think that, if possible, we should use a callback for this, instead of adding the hashes to the state manager, because:

- That way, we don't mix concerns. The blockchain has the block hashes, and the VM (through the state manager/db) access that information. Otherwise, we need to remember to always keep the DB and the blockchain in sync, which seems error-prone.
- It will be easier to do this for the forked db, because in that case using a callback is easier. With this approach, the forked DB should fetch the last 256 blocks during initialization, which is not ideal.

Reply:
I added two issues to improve this in the future. It requires a change in `revm`:
https://github.com/NomicFoundation/rethnet/issues/31
https://github.com/NomicFoundation/rethnet/issues/32

Reply:
I already implemented the changes and added them to the PR.

After I had updated a upstream dependency, the build was failing until I resolved this, so figured I might as well combine the two.